libc/sem.cc

```diff
@@ -6,16 +6,45 @@
  */
 
 #include <semaphore.h>
+#include <fcntl.h>
 #include <osv/semaphore.hh>
 #include <osv/sched.hh>
 #include <memory>
 #include "libc.hh"
+#include <unordered_map>
 
 // FIXME: smp safety
 
 struct indirect_semaphore : std::unique_ptr<semaphore> {
-    explicit indirect_semaphore(unsigned units)
-        : std::unique_ptr<semaphore>(new semaphore(units)) {}
+    private:
+        int references;
+        bool linked;
```

Comment:
Should we call it named instead of linked and reverse the boolean values? Otherwise I find it a bit confusing.

Reply:
Yeah. Maybe my suggestion was wrong in the context of my newer comments.